test/encoder.spec.js

```diff
@@ -58,6 +58,49 @@ describe('encoder', () => {
       ).to.throw()
     })
 
+    it('roundtrips an array of custom types', () => {
+      class Foo {
+        constructor (data) {
+          this.data = data
+        }
+      }
+
+      function encodeFoo (enc, foo) {
+        enc.pushAny(new cbor.Tagged(12345, foo.data))
```

Comment:
While trying to produce a fix I realized this is actually just a usage mistake, I didn't pass through the return value. Would it be worth testing the return type of external encode functions to make sure they're booleans to throw a more helpful error, or is performance too much of a concern for an extra check like that?